src/msg/consumer/config.go

```diff
@@ -30,13 +30,42 @@ import (
 
 // Configuration configs the consumer options.
 type Configuration struct {
-	Encoder                   *proto.Configuration          `yaml:"encoder"`
-	Decoder                   *proto.Configuration          `yaml:"decoder"`
-	MessagePool               *pool.ObjectPoolConfiguration `yaml:"messagePool"`
-	AckFlushInterval          *time.Duration                `yaml:"ackFlushInterval"`
-	AckBufferSize             *int                          `yaml:"ackBufferSize"`
-	ConnectionWriteBufferSize *int                          `yaml:"connectionWriteBufferSize"`
-	ConnectionReadBufferSize  *int                          `yaml:"connectionReadBufferSize"`
+	Encoder                   *proto.Configuration      `yaml:"encoder"`
+	Decoder                   *proto.Configuration      `yaml:"decoder"`
+	MessagePool               *MessagePoolConfiguration `yaml:"messagePool"`
+	AckFlushInterval          *time.Duration            `yaml:"ackFlushInterval"`
+	AckBufferSize             *int                      `yaml:"ackBufferSize"`
+	ConnectionWriteBufferSize *int                      `yaml:"connectionWriteBufferSize"`
+	ConnectionReadBufferSize  *int                      `yaml:"connectionReadBufferSize"`
+}
+
+// MessagePoolConfiguration is the message pool configuration
+// options, which extends the default object pool configuration.
+type MessagePoolConfiguration struct {
+	// Size is the size of the pool.
+	Size int `yaml:"size"`
```

Comment:
Could you just embed pool.ObjectPoolConfiguration here instead of duplicating the fields?

Reply:
Yeah I tried that, it failed to populate the fields unfortunately.